### PR TITLE
3556 after merging items, should oms retain previous merged item name on invoices, requisitions etc. 

### LINKED ISSUE #3556: After merging items, should oms retain previous merged item name on invoices, requisitions etc.?
## What went wrong? 😲
 
From [here!](https://github.com/msupply-foundation/open-msupply/issues/2450#issuecomment-2031528760) 

After merging items, should oms retain previous merged (deleted) item's names (obviously pointing to kept item afterwards) on invoice lines?

## Expected behaviour 🤔
please vote/decide:
Either, 1: retain the merged name/item for the sake of record-keeping, but point it to the kept name/item
Or, 2: once merged, no more retaining of old name/item at all; replace merged name/item with kept ones

## How to Reproduce 🔨
Steps to reproduce the behaviour:

1. Create an outbound shipment from one oms-store to another oms-store with two items
2. Sync and received the inbound and stock-in and sync
3. On desktop cloud, merge those two items and sync
4. In oms-stores, revisit the shipments: the merged-items are replaced by the kept-item in both inbound & outbound

## Your environment 🌱

- Open mSupply Version: v1.7.0
- Legacy mSupply Central Server Version: v714-dev5

- Platform: server exe, browser(chrome)

- Database type: sqlite

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3556

# 👩🏻‍💻 What does this PR do?

Continue to display the `deleted` item name in invoices (Outbound Shipment, Inbound Shipment and Prescription) instead of showing the new `merged` name.

Invoices (especially prescriptions) need to retain the value that was actually used at that time.

## 💌 Any notes for the reviewer?
Retaining the name value in requisition lines and stocktake lines will be done in separate issues

# 🧪 Testing

- [ ] Have two items with different names
- [ ] Create a couple invoices (either Outbound Shipment, Inbound Shipment, or Prescriptions) with item A
- [ ] Go to OG and merge Item A and Item B (Keep Item B)
- [ ] Sync
- [ ] Go back to the invoices you created and look at the item name column. It should point to Item A name

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/invoices/src/InboundShipment/DetailView/modals/utils.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ const createDraftInboundLine = ({
     location: undefined,
     type,
     item,
+    itemName: item.name,
```
Comment (clemens-msupply) [inline client/packages/invoices/src/InboundShipment/DetailView/modals/utils.ts:35]: So this was not populated at all previously?

File: client/packages/invoices/src/utils.ts [hunk 2]
```diff
@@ -192,7 +192,9 @@ export const inboundLinesToSummaryItems = (
     createSummaryItem(itemId, lines)
   );
 };
-export const canDeleteInvoice = (invoice: OutboundRowFragment): boolean =>
+export const canDeleteInvoice = (
+  invoice: OutboundRowFragment | PrescriptionRowFragment
+): boolean =>
```
Comment (clemens-msupply) [inline client/packages/invoices/src/utils.ts:197]: Why was this needed?

**Review by clemens-msupply — APPROVED** [review]
Looks good.
Untested though I leave testing to testers, let me know if you think I should have an extra look...

You mentioned requisitions, are they still coming or did you mean prescriptions?

Nit: shorter branch names without ",.'" please :)

File: client/packages/invoices/src/InboundShipment/DetailView/modals/utils.ts (hunk 1 above)
Comment (roxy-dao) [reply client/packages/invoices/src/InboundShipment/DetailView/modals/utils.ts:35]: Nope... some of the item related stuff aren't populated in most of the drafts.

File: client/packages/invoices/src/utils.ts (hunk 2 above)
Comment (roxy-dao) [reply client/packages/invoices/src/utils.ts:197]: I think it must have been a merge thing... Both outbound and prescriptions use this method to check if a prescription can be deleted, so need both fragments

**Comment by roxy-dao** [comment]
Requisitions and stocktake itemNames will be done in separate issues. 2 issues total (:

**Pull request merged**